src/main/java/game/map/MapValidator.java

```diff
@@ -0,0 +1,176 @@
+package game.map;
+
+import static game.map.MapHelper.getCountryWithId;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Stack;
+
+import game.pojo.Continent;
+import game.pojo.Country;
+/**
+ * Class responsible for validating the map upon loading/saving and during the editing process </br>
+ * We consider that a map is correct when it represents a connected graph, each continent
+ * also represents a connected subgraph, there are no empty continents and each country has to
+ * have a unique name and id. </br>
+ * Validation of a map is done through the method isMapValid().
+ */
+public class MapValidator {
```

Comment:
Using DFS to validate the connectivity of the map and continents looks good to me.